### ISSUE TITLE
Wrong REGEX result when pattern is a variable

### ISSUE THREAD

**Issue by tpluscode** [oxigraph/oxigraph#1024]
**Describe the bug**

`REGEX` returns always true(?) when the pattern in a variable

**To Reproduce**

1. Run Oxigraph docker v0.4.0-rc.1
2. In the UI, run the following query which should return `false`
    ```sparql
    ask {
      VALUES (?subject ?pattern) { 
        (<http://example.com/plaque/newton-s-apple-tree-monash-university> "/plaque/(?<identifier>.+)\\.html$") 
      }
      FILTER (regex(str(?subject), ?pattern, "i"))
    }
    ```
4. Compare with the expected result when pattern is inlined
    ```sparql
    ask {
      VALUES (?subject) { 
        (<http://example.com/plaque/newton-s-apple-tree-monash-university>) 
      }
      FILTER (regex(str(?subject), "/plaque/(?<identifier>.+)\\.html$", "i"))
    }
    ```

**Comment by tpluscode** [comment]
In fact, the `?pattern` can be literally anything and the result is always `true`

**Comment by Tpt** [comment]
Indeed, there is a bug in the query evaluator that makes Oxigraph consider the tested text as the regex when the regex is not constant. #1025 fixes that. Thank you for spotting it!

**Comment by tpluscode** [comment]
You're welcome. And as always, thanks for a quick fix

**Comment by Tpt** [comment]
Should be fixed in v0.4.1

**Comment by tpluscode** [comment]
> Should be fixed in v0.4.1

I confirm it is. Thank you
